catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -229,10 +230,28 @@ public boolean sourceStatus(String servicePID) {
                             source.put(MAP_ENTRY_NAME, config.getPid());
                         }
 
-                        Dictionary<String, Object> properties = config.getProperties();
+                        Dictionary<String, Object> configProperties = config.getProperties();
+                        List<Map<String, Object>> metatypeProperties =
+                                (List<Map<String, Object>>) metatype.get("metatype");
                         Map<String, Object> plist = new HashMap<>();
-                        for (String key : Collections.list(properties.keys())) {
-                            plist.put(key, properties.get(key));
+                        Object propertyValue;
+                        for (String key : Collections.list(configProperties.keys())) {
+                            propertyValue = configProperties.get(key);
+                            // If the configuration property is a password, set its value to
+                            // "password" if it's not empty, so that the real password value will
+                            // be hidden. (Passwords within source configs are usually optional,
+                            // thus it should have the ability to update to an empty string).
+                            for (Map<String, Object> metatypeProperty : metatypeProperties) {
+                                if (metatypeProperty.get("id")
+                                        .equals(key) &&
+                                        AttributeDefinition.PASSWORD
+                                                == (Integer) metatypeProperty.get("type")
+                                        && !"".equals(propertyValue)) {
```

Comment:
You're looping unnecessarily over anything that has a `propertyValue` of an empty string. This whole nested loop can be rewritten much more efficiently as something like...

```java
                        while (configProperties.keys()
                                .hasMoreElements()) {
                            String key = configProperties.keys()
                                    .nextElement();
                            propertyValue = configProperties.get(key);
                            if (!propertyValue.equals("")) {
                                for (Map<String, Object> metatypeProperty : metatypeProperties) {
                                    if (metatypeProperty.get("id")
                                            .equals(key) && AttributeDefinition.PASSWORD
                                            == (Integer) metatypeProperty.get("type")) {
                                        propertyValue = "password";
                                        break;
                                    }
                                }
                            }
                            plist.put(key, propertyValue);
                        }
```